### PR TITLE
Navigation links v2 152718702

### PR DESCRIPTION
**Pull request by snehanagole-osi**
### What is Changed:
 - Renamed SideNav to SideBar.
 - Refactor SideBar, NavLinks to be Generic components. These components now render given 
    children.
 - Refactor Layout03 component to take SideBar properties & pass them down to SideBar component.
 - NavLinksContainer component has been deleted. It has been converted to a story in storybook. 
   This demonstrates how to implement a NavLinks Active Highlighter. Note: This does NOT 
   demonstrate complete ScrollSpy functionality.
 - SideBar, NavLinks, NavLink components are not in beta state anymore.
 - Refactored/Added tests to use Enzyme.

### What is New:
 - Link, PreIcon, PostIcon components. These are used in NavLink component.
 - Using 'classnames' library to simplify className construction.

### CODE REVIEW

**Comment by edgibbs** [comment]
Can one of the admins verify this patch?

**Comment by edgibbs** [comment]
retest this please

**Comment by edgibbs** [comment]
Build triggered. sha1 is merged.

**Comment by edgibbs** [comment]
Build started sha1 is merged.

**Comment by edgibbs** [comment]
Everything is green!
 No test results found.

File: README.md [hunk 1]
```diff
@@ -37,9 +37,12 @@ The **react-wood-duck component library** is a set of re-usable components and s
 	* DateTimePicker
 	* InputComponent
 	* ListItem
+        * Link
```
Comment (edgibbs) [inline README.md]: Like the names of these components.

File: src/Link.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import classNames from 'classnames';
+
+const Link = function(props) {
+  const { text, href, active, clickHandler } = props;
+  const classes = classNames({ link: true, active: active });
+  return (
+    <a
+      href={href}
+      className={classes}
+      onClick={e => clickHandler(props.href, e)}
```
Comment (edgibbs) [inline src/Link.js]: prefer full names even here like event over e

**Comment by edgibbs** [comment]
retest this please

**Comment by edgibbs** [comment]
Build started sha1 is merged.

**Comment by edgibbs** [comment]
Everything is green!
 No test results found.

File: src/test/NavLinks_test.js [hunk 3]
```diff
@@ -1,168 +1,108 @@
...
 import { shallow } from 'enzyme';
 import './EnzymeSetup';
-
 import NavLink from '../NavLink';
 import NavLinks from '../NavLinks';
 
-describe('NavLinks', () => {
+describe('NavLinks', function() {
+  const clickHandler = function() {
+    console.log('test msg');
```
[the remark this answers is not in the document]
Comment (snehanagole-osi) [reply src/test/NavLinks_test.js]: Removed it.

**Comment by snehanagole-osi** [comment]
@edgibbs : changes done based on your feedback. Can you please review them.

**Comment by edgibbs** [comment]
retest this please

**Comment by edgibbs** [comment]
Build triggered. sha1 is merged.

**Comment by edgibbs** [comment]
Build started sha1 is merged.

**Comment by edgibbs** [comment]
Everything is green!
 No test results found.

**Comment by edgibbs** [comment]
@alyssaburkecwds Oked the design for this iteration.

**Comment by edgibbs** [comment]
retest this please

**Comment by edgibbs** [comment]
Build triggered. sha1 is merged.

**Comment by edgibbs** [comment]
Build started sha1 is merged.

**Comment by edgibbs** [comment]
Everything is green!
 No test results found.

**Pull request merged**